src/client/datascience/notebook/kernelProvider.ts

```diff
@@ -119,7 +132,20 @@ export class VSCodeKernelPickerProvider implements NotebookKernelProvider {
         );
         this.notebook.onDidChangeActiveNotebookKernel(this.onDidChangeActiveNotebookKernel, this, disposables);
     }
-    @captureTelemetry(Telemetry.KernelProviderPerf)
+
+    public get onResolvedKernel() {
+        return this._onDidResolveKernel.event;
+    }
+
+    public async resolveKernel?(
+        kernel: VSCodeNotebookKernelMetadata,
+        document: NotebookDocument,
+        webview: NotebookCommunication,
+        _token: CancellationToken
+    ): Promise<void> {
+        this._onDidResolveKernel.fire({ kernel, document, webview });
```

Comment:
Hmm, this is supposed to wait for the the UI kernel to get initialized.
Now we're not waiting here, its just a fire & forget.
I'm concerened this could introduce race conditions.

Also resolveKernel is not really an event, the method is VS Code telling the extension host to resolve the extension. Its a request as opposed to  callback/notification.
More like a request to initialize the kernel on the UI side (webview).
Hence having an event named `onDidResolveKernel` is not correct (as the kernel has not been resolved).